backends/p4tools/modules/testgen/options.cpp

```diff
@@ -382,4 +383,15 @@ TestgenOptions::TestgenOptions()
         "produce no tests or only tests that contain counter examples.");
 }
 
+void TestgenOptions::validate() const {
+    if (minCoverage > 0 && !hasCoverageTracking) {
+        ::error(
+            ErrorType::ERR_INVALID,
+            "It is not allowed to have --assert-min-coverage set to non-zero without a coverage "
+            "tracking enabled with --track-coverage option. Without coverage tracking, the "
+            "--assert-min-coverage is meaningless.");
+        FATAL_ERROR("Incompatible options given.");
```

Comment:
I approved but I do not think this is necessary. We will exist if we encounter an error during processing. If not, that is a bug.

Reply:
Hmm that is a little odd, we have been using `::error` just fine in the options classes and the compiler context is set up before the validate call.

The entire way options are handled in P4Testgen is a little funky (there are a bunch of hacks to set up context). So there  might be something awry there.

Reply:
Maybe the exception thrown with FATAL_ERROR causes the problem? We have not tested any exception handling that early.

Reply:
I think the handler for errors in the option parsing works differently, it also prints the usage if the argument parsing fails. What I was testing (and what produced this weird error) was just having the `error` there and no `FATAL_ERROR`, so there were no exceptions thrown.

Reply:
Did this get resolved?

Reply:
Unless we change error handling during process, the FATAL_ERROR is the best bet to cleanly end the processing (in the compiler, there are only warnings, so this does not matter).

Reply:
I looked into this, basically we need to do this 
```
if (::errorCount() > 0) {
    return std::nullopt;
}
``` 
after the validate call to not return a broken compile context. We could also return  true/false in validate.

Reply:
Made it return. `::errorCount()` is not available before the context is set (it could be extracted from the context, but this seems cleaner to me).